### PR TITLE
Fix stream join

### LINKED ISSUE #956: bug:  index name need quote
error stack:

```
(base) ➜  oomcli git:(fix-stream-join) ✗ DEBUG=1 oomcli join --feature user-click.last_5_click_posts --input-file /tmp/oomstore/entity_rows.csv --output csv
Error: failed joining historical features: pq: syntax error at or near "-"
github.com/oom-ai/oomstore/pkg/errdefs.WithStack
	/Users/lianxm/github.com/oomstore/pkg/errdefs/wrap.go:33
github.com/oom-ai/oomstore/internal/database/dbutil.(*DBOpt).ExecContext
	/Users/lianxm/github.com/oomstore/internal/database/dbutil/db_opt.go:37
github.com/oom-ai/oomstore/internal/database/offline/sqlutil.PrepareJoinedTable
	/Users/lianxm/github.com/oomstore/internal/database/offline/sqlutil/join_helper.go:113
github.com/oom-ai/oomstore/internal/database/offline/sqlutil.JoinOneGroup
	/Users/lianxm/github.com/oomstore/internal/database/offline/sqlutil/join.go:119
github.com/oom-ai/oomstore/internal/database/offline/sqlutil.DoJoin
	/Users/lianxm/github.com/oomstore/internal/database/offline/sqlutil/join.go:81
github.com/oom-ai/oomstore/internal/database/offline/sqlutil.Join
	/Users/lianxm/github.com/oomstore/internal/database/offline/sqlutil/join.go:44
github.com/oom-ai/oomstore/internal/database/offline/postgres.(*DB).Join
	/Users/lianxm/github.com/oomstore/internal/database/offline/postgres/store.go:42
github.com/oom-ai/oomstore/pkg/oomstore.(*OomStore).ChannelJoin
	/Users/lianxm/github.com/oomstore/pkg/oomstore/join.go:55
github.com/oom-ai/oomstore/oomcli/cmd.join
	/Users/lianxm/github.com/oomstore/oomcli/cmd/join_helper.go:27
github.com/oom-ai/oomstore/oomcli/cmd.glob..func24
	/Users/lianxm/github.com/oomstore/oomcli/cmd/join.go:25
github.com/spf13/cobra.(*Command).execute
	/Users/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:860
github.com/spf13/cobra.(*Command).ExecuteC
	/Users/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974
github.com/spf13/cobra.(*Command).Execute
	/Users/lianxm/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/oom-ai/oomstore/oomcli/cmd.Execute
	/Users/lianxm/github.com/oomstore/oomcli/cmd/root.go:32
main.main
	/Users/lianxm/github.com/oomstore/oomcli/main.go:6
runtime.main
	/Users/lianxm/local/go/src/runtime/proc.go:255
runtime.goexit
	/Users/lianxm/local/go/src/runtime/asm_amd64.s:1581
```

 This is because pg does not support index with the character '-' in it.

### PR DESCRIPTION
**Pull request by lianxmfor**
close #956

### CODE REVIEW

File: pkg/oomstore/join.go [hunk 1]
```diff
@@ -24,23 +24,13 @@ func (s *OomStore) ChannelJoin(ctx context.Context, opt types.ChannelJoinOpt) (*
...
 	if err != nil {
 		return nil, err
 	}
 
-	features = features.Filter(func(f *types.Feature) bool {
-		return f.Group.Category == types.CategoryBatch
-	})
-	if len(features) == 0 {
-		return emptyResult, nil
-	}
```
Comment (jinghancc) [inline pkg/oomstore/join.go:42]: in case user pass in empty feature list, we could keep the empty check

**Review by jinghancc — APPROVED** [review]
LGTM

File: pkg/oomstore/join.go (hunk 1 above)
Comment (lianxmfor) [reply pkg/oomstore/join.go:42]: good catch!

**Review by wfxr — APPROVED** [review]
lgtm

**Pull request merged**